File: src/main/java/fi/eonwe/wikilinks/fatpages/WikiPage.java
package fi.eonwe.wikilinks.fatpages;

/**
 */
public interface WikiPage {

    /**
     * Is this a redirect to another page or an actual page?
     * @return
     */
    boolean isRedirect();

    /**
     * @return a unique identifier for this page
     */
    int getId();

    /**
     * @return
     */
    String getTitle();

}

File: src/main/kotlin/fi/eonwe/wikilinks/fatpages/WikiPage.kt
package fi.eonwe.wikilinks.fatpages

/**
 */
interface WikiPage {
    /**
     * Is this a redirect to another page or an actual page?
     * @return
     */
    val isRedirect: Boolean

    /**
     * @return a unique identifier for this page
     */
    fun id(): Int

    /**
     * @return title of the page
     */
    fun title(): String
}